### DISCUSSION TITLE
Can no longer change target directory

### DISCUSSION THREAD

**Discussion by pmruzicka** [aclap-dev/video-downloadhelper#814] [Help and Q&A]
### Your question

Link to it does nothing on new version (which I hate).  I can't find where my downloads are going. And I do NOT want it set to the default. 

### Addon & CoApp version and details

version: 9.0.2.3
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0
{
    "theme": "dark",
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 5,
    "successfull_dl": 3
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

Refer to #665.  I'm sure you hate the new version only because you have not yet discovered how to use it properly.  There isn't a proper user manual, only the Wiki, whose link is at the top of every page of this forum.  You might get some help from reading the Wiki.  But in the end, you are left to experiment on your own & just discover how it works by trial & error.  Don't be put off so soon.  Give it some time.  I believe the new version is a quantum leap forward.  For one thing, you can now operate VDH strictly from the keyboard.  Click click click with the mouse is no longer forced upon you.  You can get at pretty much everything using keyboard operations.  For me, that's worth any other adapting I have to do.

**Reply by PeBoVision** [reply]
I replied elsewhere that your suggestions worked until they didn't (I lost "downlaod as" and the main buttons remained "Download".
Well, I uninstalled and reinstalled VDH, followed you instructions on a fresh install, and this time "Remember These Settings worked, and the "download" button became a "download as" and I am prompted for each download, and even surviving a reboot.
From one grandpa to another, Merci Papère!

**Comment by MediumUK** [comment]
I agree with pmruzicka!
 
Extremely frustrating FORCED STUPID default download directory - with NO discernable way to change the download directory!

Even the "SHOW DOWNLOAD DIRECTORY" does NOTHING (if the last download directory setting can't be readily recalled)!

This is an unbelievably ridiculous usability nightmare!

**Reply by paulrouget** [reply]
Hello. It doesn't hurt to be polite when posting online. I am usually happy fixing bugs, but it's a lot easier when people are being patient and courteous.

We have a code of conduct: https://github.com/aclap-dev/video-downloadhelper/wiki/Code-Of-Conduct and you managed to break the only 3 rules there at once :)

Be nice please.

**Reply by Kostenzer** [reply]
@MediumUK
I couldn't believe what you wrote about VDH. And then the language you use. You seem to have a small vocabulary.
If you then also write "an unbelievably ridiculous usability nightmare!"
I would say, use another add-on for this purpose or program a perfect program yourself. And yes, the program could be a bit better when it comes to making default settings. It used to be a bit better, as far as I can remember. But if you report that in a normal way, it is seen as constructive criticism. You achieve more with that.
VDH programmers, you are doing well. I would like to tell you, block such people who act like this.

**Comment by GrampaWildWilly** [comment]
You folks are just not being adventurous enough.

![#001](https://github.com/user-attachments/assets/1cea1177-b4ca-4ee9-a2b7-5dbaea50c463)
![#002](https://github.com/user-attachments/assets/a75cf6e0-96cf-4c71-a9f1-5a72ad51bc94)
![#003](https://github.com/user-attachments/assets/1c8ee78d-abce-44bf-a3e0-bbddacc8d263)

So you would change the `Download directory` before you launch your download.  Then VDH would use whatever you set there.  But that's only if you use the `Download` function.  This process is entirely too tedious & you should not be using it.  Here's a better way.

![#004](https://github.com/user-attachments/assets/b7df4677-e3bc-442f-bc3c-f8c9b3b52fbb)
![#005](https://github.com/user-attachments/assets/8431227f-4625-407e-ac9c-2fa5ea1a4d33)
![#006](https://github.com/user-attachments/assets/8c5f3c60-5479-45d5-968f-51f6c58cb0d8)
![#007](https://github.com/user-attachments/assets/1a6daf2d-dc60-46c8-bf2f-19b9759bc7f4)
![#008](https://github.com/user-attachments/assets/49001206-7d64-4e4d-8862-a9825fdb8cc1)

All of that is not terribly different from the way VDH has worked in past releases.  You really should have been able to figure this stuff out on your own.  I did.  There is no proper documentation of this.  Well, I suppose I just supplied it.  But I figured most of this out by just clicking.  Asking what does that do?  What happens if I click this?  What's inside this?  Like I say, you need to be more adventurous.

Once you have changed your default action to `Download As...` you will always get the file selection dialog.  VDH remembers 2 most recent target directories.  It remembers the one in the settings, which it uses when you do `Download` functions.  Separately, it remembers the one you used in your most recent `Download As...` operation.  So you can `Download As...` into the same directory repeatedly on consecutive operations.  Of course, you can switch to another directory whenever you want.  That will become the new default for your subsequent `Download As...` operations.  I emphasize, these are 2 separate most recent download directories.  Each `Download As...` operation DOES NOT change the `Download directory` in the VDH settings.

Now, the question often comes up, "Where did that download go?"  That can be answered by VDH itself.  Go into the VDH settings again & click the `Copy information to clipboard` button you can see in my third image above.  After you click that, immediately switch tasks to your text editor & paste from your system clipboard into the text editor.  Do a Shift+Insert.  I've attached mine here:

[VDH Information.txt](https://github.com/user-attachments/files/16276439/VDH.Information.txt)

Look in there for a line that starts with `"download_directory":` & another line that starts with `"__serializer_value":`.  Just to be confusing, there's 2 lines with `"__serializer_value":`.  Look at the second one.  Those are 2 directories on my system.  The first one is the `Download directory` in my VDH settings.  The second one is a different directory because that's where I did my most recent `Download As...` into.  If you're wondering where your download went, it's probably one of those 2 directories.  Now, if you've done a bunch of downloads into a bunch of different directories & you've lost track, this may not give you your answer.  I can only recommend that if you're doing that sort of scattershot downloading, you need to keep careful track of what you're doing using your own methods outside VDH.  VDH can keep track of only the 2 directories.  If you think VDH ought to be keeping a history log of its activity, I invite you to join the long list of people who have asked for that over the years.  We're still waiting for that one.
 
As for the `Show download folder` function, this is somewhat hamstrung by shortcomings in the Windows File Explorer.  I'm getting the impression the File Explorer is still a work in progress.  The most recent Microsoft system maintenance did make some changes to it.  I am waiting for them to fix it so you can execute explorer.exe & provide several directory specifications as parameters.  That ought to cause the file manager window to open with several tabs, one for each directory you specified.  It doesn't work that way now, but I have fervent hopes they will get their act together & provide that functionality.

In the meantime, the `Show download folder` button in VDH is somewhat at the mercy of the currently under-featured Windows file manager.  I always have the file manager open with a dozen tabs.  When I click the button in the VDH menu, focus switches to the file manager.  That's all.  It goes to the file manager, to whatever tab you happen to have in focus.  It DOES NOT go to the tab where I have the directory VDH downloaded that video into . . . unless I accidentally happen to have that tab in focus in the file manager window.

VDH DOES NOT cause a new file manager instantiation to open on the right directory.  It should.  I have at least one other application on my system that does launch a new instantiation of the file manager when you execute a certain function within that application.  So it is possible to do despite the Windows file manager leaving much to be desired today.  VDH just doesn't do it at the moment.  @paulrouget Paul should know about this.  If he doesn't, we just told him.  Let's hope he eventually gets around to fixing this.

If the file manager had better APIs, VDH would be able to query an open file manager window & find out if the right directory happens to be a tab somewhere within that window.  There should then be another API that tells that existing instantiation of the file manager to tab over to the desired directory.  If that directory doesn't happen to already be a tab in that window, there should be another API that adds our VDH download target directory as a new tab in the existing file manager window.  It should be an option within VDH to either add a new tab to an existing instantiation of the file manager or open a whole new file manager window on the desired directory.  But like I say, I don't think those file manager APIs exist today.  Maybe some day . . .

**Reply by jabberwockxeno** [reply]
If I click "change" here, nothing happens.

**Reply by GrampaWildWilly** [reply]
Didn't notice this comment 3 weeks ago.  In any case, you should not be trying to change the download directory by clicking the `Change` button in the VDH settings.  You should be using `Download As` instead.  Read what I explained above.  Follow my advice.  You won't care whether the `Change` button does anything.

**Comment by paulrouget** [comment]
So that was just a case of people not finding the setting to change the Download Directory?

Please re-open if I'm wrong.

**Reply by jabberwockxeno** [reply]
I don't know how to reopen it, but that is not the issue for me: Clicking the "change" button/link simply does nothing

**Reply by PeBoVision** [reply]
Ignore my last comment.
Closed Firefox, and when I reopened it the "download as" no longer appears...so again, no way to download. It goes through the motions showing a progress bar and everything but no file is saved anywhere on my computer.
I guess I have to wait for a coapp update. Until then, I'll drop back to Windows if I want to save a video with VDH.

Too bad, I was really pleased to find "download As" and now it's gone. A very frustrating plug-in.

**Comment by PeBoVision** [comment]
Clicking More settings brought up a VERY different menu for me, only providing an option for selecting dark or light themes.
                                                                                 - BUT - 
I had no idea that the gear button on the main menu brings up a different set of options when a video is playing in the browser. I kept trying to fix things BEFORE playing a video, and assumed that was the only menu the gear linked to. Finding "save as" and "Remember this setting" was the ticket and got everything working.

Merci ben PapèreWillySauvage! Tes liens ont donné la bonne réponse! Go! Habs! Go!
